### PR TITLE
Remove local from pom and package names

### LINKED ISSUE #2810: Remove use of world 'local' in poms and classes
`	<artifactId>spring-cloud-dataflow-server-local-autoconfig</artifactId>` should change to `single-autoconfig`

and classes prefixed with `local` should be changed to `single`.

### PR DESCRIPTION
**Pull request by jvalkeal**
- Rename spring-cloud-dataflow-server-local-autoconfig to spring-cloud-dataflow-autoconfigure
  which would be the one where all boot related autoconfiguration classess should eventually go.
- Rename spring-cloud-starter-dataflow-server-local to spring-cloud-starter-dataflow-server as
  we're really only building one server.
- Fixes #2810

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -72,7 +72,7 @@
 		<module>spring-cloud-dataflow-registry</module>
 		<module>spring-cloud-dataflow-rest-resource</module>
 		<module>spring-cloud-dataflow-server-core</module>
-		<module>spring-cloud-dataflow-server-local-autoconfig</module>
+		<module>spring-cloud-dataflow-autoconfigure</module>
```
Comment (ilayaperumalg) [inline pom.xml]: Can we make this as `spring-cloud-dataflow-server-autoconfigure`?

Comment (jvalkeal) [reply pom.xml]: Point is that my hope is that eventually, maybe somewhere in 2.1.x line we have one autoconfigure jar instead of 7 we use now.

Comment (jvalkeal) [reply pom.xml]: We need to stop doing this:
```
$ grep EnableAutoConfiguration */src/main/resources/META-INF/spring.factories
spring-cloud-dataflow-configuration-metadata/src/main/resources/META-INF/spring.factories:org.springframework.boot.autoconfigure.EnableAutoConfiguration:\
spring-cloud-dataflow-platform-cloudfoundry/src/main/resources/META-INF/spring.factories:org.springframework.boot.autoconfigure.EnableAutoConfiguration=\
spring-cloud-dataflow-platform-kubernetes/src/main/resources/META-INF/spring.factories:org.springframework.boot.autoconfigure.EnableAutoConfiguration=\
spring-cloud-dataflow-rest-client/src/main/resources/META-INF/spring.factories:org.springframework.boot.autoconfigure.EnableAutoConfiguration=\
spring-cloud-dataflow-server-core/src/main/resources/META-INF/spring.factories:org.springframework.boot.autoconfigure.EnableAutoConfiguration=\
spring-cloud-dataflow-server-local-autoconfig/src/main/resources/META-INF/spring.factories:org.springframework.boot.autoconfigure.EnableAutoConfiguration=\
spring-cloud-dataflow-shell-core/src/main/resources/META-INF/spring.factories:org.springframework.boot.autoconfigure.EnableAutoConfiguration=\
```
Every separate jar makes app startup slower.

**Comment by ilayaperumalg** [comment]
Rebased and merged as 9c873735

**Pull request closed without merging**